### PR TITLE
AWS Provider 3.x and other fixes

### PR DESCRIPTION
**Pull request by brikis98**
AWS Provider 3.0.0 just came out and has a bunch of backwards incompatible changes. This updates this repo to work with 3.x.

### CODE REVIEW

**Comment by brikis98** [comment]
Hm, a couple tests are failing repeatedly... Mostly from not finding the expected number of instances in the cluster: 5 instead of 6, or, even more weirdly, 12 instead of 6... Some sort of issue with multiple clusters sometimes being deployed in the same region and the tags not being unique enough? Will not have time to dig into this today, so I'll have to come back to it. If anyone has some cycles to spend on this and dig in, that would be most appreciated.

File: examples/nomad-consul-ami/nomad-consul-docker.json [hunk 1]
```diff
@@ -4,12 +4,13 @@
     "aws_region": "us-east-1",
     "nomad_version": "0.9.1",
     "consul_module_version": "v0.7.4",
-    "consul_version": "1.5.1"
+    "consul_version": "1.5.1",
+    "ami_name_prefix": "nomad-consul"
```
Comment (rhoboat) [inline examples/nomad-consul-ami/nomad-consul-docker.json:8]:
```suggestion
    "ami_name_prefix": "nomad-consul",
```

File: examples/nomad-consul-ami/nomad-consul.json [hunk 2]
```diff
@@ -4,12 +4,13 @@
     "aws_region": "us-east-1",
     "nomad_version": "0.9.1",
     "consul_module_version": "v0.7.4",
-    "consul_version": "1.5.1"
+    "consul_version": "1.5.1",
+    "ami_name_prefix": "nomad-consul"
```
Comment (rhoboat) [inline examples/nomad-consul-ami/nomad-consul.json:8]:
```suggestion
    "ami_name_prefix": "nomad-consul",
```

**Comment by brikis98** [comment]
Thx for the review!

**Pull request merged**